src/utils/daos_hdlr.c

```diff
@@ -1568,11 +1600,1347 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+			src_file_dfs->offset = 0;
+			dst_file_dfs->offset = 0;
+
+			/* set perms on files to original source perms */
+			rc = file_chmod(src_file_dfs, filename, st.st_mode);
+			if (rc != 0) {
+				fprintf(stderr, "updating source file
+					permissions failed (%d)\n", rc);
+				D_GOTO(out, rc);
+			}
+			rc = file_chmod(dst_file_dfs, dst_filename, st.st_mode);
+			if (rc != 0) {
+				fprintf(stderr, "updating dst file
+					permissions failed (%d)\n", rc);
+				D_GOTO(out, rc);
+			}
+
+			/* close src and dst */
+			file_close(src_file_dfs, filename);
+			file_close(dst_file_dfs, filename);
+		} else if (S_ISDIR(st.st_mode)) {
+			/* Check that the directory is not "src_dir"
+			 * or src_dirs's parent.
+			 */
+			if ((strcmp(d_name, "..") != 0) &&
+				(strcmp(d_name, ".") != 0)) {
+				char path[MAX_FILENAME];
+				char dpath[MAX_FILENAME];
+
+				path_length = snprintf(path, MAX_FILENAME, "%s",
+							filename);
+				if (path_length >= MAX_FILENAME) {
+					rc = ENAMETOOLONG;
+					fprintf(stderr, "Path length is too
+						long on source.\n");
+					D_GOTO(out, rc);
+				}
+				path_length = snprintf(dpath, MAX_FILENAME,
+							"%s", dst_filename);
+				if (path_length >= MAX_FILENAME) {
+					rc = ENAMETOOLONG;
+					fprintf(stderr, "Path length is too
+						long on destination.\n");
+					D_GOTO(out, rc);
+				}
+
+				mode_t tmp_mode_dir = S_IRWXU;
+
+				rc = file_mkdir(dst_file_dfs, dpath,
+						&tmp_mode_dir);
+				/* continue if directory already exists,
+				 * fail otherwise
+				 */
+				if (rc != EEXIST && rc != 0) {
+					D_GOTO(out, rc);
+				}
+				/* Recursively call "fs_copy"
+				 * with the new path.
+				 */
+				rc = fs_copy(src_file_dfs, dst_file_dfs, path,
+						dfs_prefix_len, fs_dst_prefix);
+				if (rc != 0) {
+					fprintf(stderr, "filesystem copy
+						failed, %d.\n", rc);
+					D_GOTO(out, rc);
+				}
+
+				/* set original source perms on directories
+				 * after copying
+				 */
+				rc = file_chmod(src_file_dfs, path, st.st_mode);
+				if (rc != 0) {
+					fprintf(stderr, "updating source
+						permissions failed on %s
+						(%d)\n", path, rc);
+					D_GOTO(out, rc);
+				}
+				rc = file_chmod(dst_file_dfs, dpath,
+						st.st_mode);
+				if (rc != 0) {
+					fprintf(stderr, "updating destination
+						permissions failed on %s
+						(%d)\n", dpath, rc);
+					D_GOTO(out, rc);
+				}
+
+			} else {
+				/* if this is src_dir or src_dir'S parent
+				 * continue to next entry if there is one
+				 */
+				continue;
+			}
+		} else {
+			rc = ENOTSUP;
+			fprintf(stderr, "file type is not supported (%d)\n",
+				rc);
+			D_GOTO(out, rc);
+		}
+	}
+
+	/* After going through all the entries, close the directory. */
+	if (file_closedir(src_file_dfs, src_dir))
+		fprintf(stderr, "Could not close '%s': %s\n",
+			dir_name, strerror (errno));
+out:
+	return rc;
+}
+
+static int
+fs_copy_connect(struct file_dfs *src_file_dfs,
+		struct file_dfs *dst_file_dfs,
+	   	struct fs_copy_args *fa,
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs